service/session_http.go

```diff
@@ -0,0 +1,160 @@
+		return
+	}
+	return
+}
+
+//userLoginHandler function take credentials in json
+// and check if the credentials are correct
+// also generate and returns a JWT token in the case of correct crendential
+func userLoginHandler(deps Dependencies) http.HandlerFunc {
+	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		user := db.User{}
+
+		//fetching the json object to get crendentials of users
+		err := json.NewDecoder(req.Body).Decode(&user)
+		if err != nil {
+			logger.WithField("err", err.Error()).Error("JSON Decoding Failed")
+			rw.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		//TODO change no need to return user object from Authentication
+		//checking if the user is authenticated or not
+		// by passing the credentials to the AuthenticateUser function
+		user, err1 := deps.Store.AuthenticateUser(req.Context(), user)
+		if err1 != nil {
+			logger.WithField("err", err1.Error()).Error("Invalid Credentials")
+			rw.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+
+		//Generate new JWT token if the user is authenticated
+		// and return the token in request header
+		token, err := generateJwt(user.ID)
+		if err != nil {
+			rw.WriteHeader(http.StatusInternalServerError)
+			rw.Write([]byte("Token Generation Failure"))
+			return
+		}
+		authbody := AuthBody{
+			Message: "Login Successfull",
+			Token:   token,
+		}
+
+		respBytes, err := json.Marshal(authbody)
+		if err != nil {
+			ae.Error(ae.ErrJSONParseFail, "JSON Parsing Failed", err)
+		}
+
+		rw.Header().Add("Content-Type", "application/json")
+		rw.Write(respBytes)
+	})
+}
+
+//userLogoutHandler function logs the user off
+// and add the valid JWT token in BlacklistedToken
+func userLogoutHandler(deps Dependencies) http.Handler {
+	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+
+		//fetching the token from header
+		authToken := req.Header["Token"]
+
+		//fetching details from the token
+		userID, expirationTimeStamp, err := getDataFromToken(authToken[0])
+		if err != nil {
+			rw.WriteHeader(http.StatusUnauthorized)
+			rw.Write([]byte("Unauthorized"))
+			return
+		}
+		expirationDate := time.Unix(expirationTimeStamp, 0)
+
+		//create a BlacklistedToken to add in database
+		// To blacklist a user valid token
+		userBlackListedToken := db.BlacklistedToken{
+			UserID:         userID,
+			ExpirationDate: expirationDate,
+			Token:          authToken[0],
+		}
+
+		err = deps.Store.CreateBlacklistedToken(req.Context(), userBlackListedToken)
+		if err != nil {
+			ae.Error(ae.ErrFailedToCreate, "Error creating blaclisted token record", err)
+			rw.Header().Add("Content-Type", "application/json")
+			ae.JSONError(rw, http.StatusInternalServerError, err)
+			return
+		}
+
+		rw.Write([]byte("Logged Out Successfully"))
+		rw.WriteHeader(http.StatusOK)
+		return
+	})
+}
+
+func getDataFromToken(Token string) (userID float64, expirationTime int64, err error) {
```

Comment:
move `getDataFromToken` to token package

example
token.Parse(token string) (userID float64, expirationTime int64, err error)